bottlecap/src/traces/trace_agent.rs

```diff
@@ -298,6 +325,108 @@ impl TraceAgent {
             .body(Body::from(response_json.to_string()))
     }
 
+    /// Generic proxy handler for forwarding requests to Datadog backends
+    async fn handle_proxy(
+        config: Arc<config::Config>,
+        req: Request<Body>,
+        backend_domain: &str,
+        backend_path: &str,
+        error_context: &str,
+    ) -> http::Result<Response<Body>> {
+        let (parts, body) = req.into_parts();
+
+        // TODO: Update this when upgrading hyper
+        #[allow(deprecated)]
+        let body_bytes = match hyper::body::to_bytes(body).await {
+            Ok(bytes) => bytes,
+            Err(err) => {
+                return log_and_create_http_response(
+                    &format!("Error reading request body: {err}"),
+                    StatusCode::INTERNAL_SERVER_ERROR,
+                );
+            }
+        };
+
+        let target_url = format!("https://{}.{}{}", backend_domain, config.site, backend_path);
+        let client = http_client::get_client(config.clone());
+        let mut request_builder = client.post(&target_url);
+
+        for (name, value) in &parts.headers {
+            if name.as_str().to_lowercase() != "host"
+                && name.as_str().to_lowercase() != "content-length"
+            {
+                if let Ok(header_value) =
+                    reqwest::header::HeaderValue::from_str(value.to_str().unwrap_or_default())
+                {
+                    request_builder = request_builder.header(name.as_str(), header_value);
+                }
+            }
+        }
+        request_builder = request_builder.header("DD-API-KEY", &config.api_key);
+        request_builder = request_builder.header(
+            "X-Datadog-Additional-Tags",
+            format!(
+                "_dd.origin:lambda;functionname:{}",
```

Comment:
Should we add then all the other tags that the tags provider has?

Reply:
Hmm. We could, but the docs sheet I was provided doesn't ask for them so I'm not inclined to add extra tags if they aren't used by those products.

We can always add them later.